modules/stereo/src/descriptor.hpp

```diff
@@ -0,0 +1,451 @@
+            parallelSymetricCensus(cv::Mat img[2], int k2,int numberOfImages, int *distance[2])
+            {
+                if(numberOfImages == SingleImageDescriptor)
+                {
+                    image[0] = img[0].data;
+                    dst[0] = distance[0];
+                    imgStop = 0;
+                }
+                else
+                {
+                    image[1] = img[1].data;
+                    dst[1] = distance[1];
+                    imgStop = 1;
+                }
+                n2 = k2;
+                width = img[1].cols;
+                height = img[1].rows;
+            }
+            void operator()(const cv::Range &r) const {
+                for (int i = r.start; i <= r.end ; i++)
+                {
+                    int distV = (i)* width;
+                    int c[2];
+                    for (int j = n2; j <= width - n2; j++)
+                    {
+                        for(int d = 0; d <= imgStop; d++)
+                        {
+                            c[d] = 0;
+                            //the classic center symetric census which compares the curent pixel with its symetric not its center.
+                            for (int ii = -n2; ii <= 0; ii++)
+                            {
+                                int rrWidth = (ii + i) * width;
+                                for (int jj = -n2; jj <= +n2; jj++)
+                                {
+                                    if (image[d][(rrWidth + (jj + j))] > image[d][((ii * (-1) + i) * width + (-1 * jj) + j)])
+                                    {
+                                        c[d] = c[d] + 1;
+                                    }
+                                    c[d] = c[d] * 2;
+                                    if(ii == 0 && jj < 0)
+                                    {
+                                        if (image[d][(i * width + (jj + j))] > image[d][(i * width + (-1 * jj) + j)])
+                                        {
+                                            c[d] = c[d] + 1;
+                                        }
+                                        c[d] = c[d] * 2;
+                                    }
+                                }
+                            }
+                            dst[d][(distV + j)] = c[d];
+                        }
+                    }
+                }
+            }
+        };
+        /**
+        Two variations of census applied on input images
+        Implementation of a census transform which is taking into account just the some pixels from the census kernel thus allowing for larger block sizes
+        **/
+        //void applyCensusOnImages(const cv::Mat &im1,const cv::Mat &im2, int kernelSize, cv::Mat &dist, cv::Mat &dist2, const int type);
+        void applyCensusOnImages(const cv::Mat &image1, const cv::Mat &image2, int kernelSize, cv::Mat dist[2], const int type,const int numImages = DoubleImageDescriptor);
```

Comment:
The naming in public API should be terse and consistent. 

- don't use words, like `apply` or `compute` in the beginning
- don't use abbreviations, like MCT
- don't use the `onImages` suffix - the APIs are different, so it will work with overloading

The good names are `censusTransform`, `modifiedCensusTransform`, etc.  

Also, please, use references (&) everywhere you can (e.g. for output Mats). It doesn't make much difference, but is consistent with other functions.